vllm/v1/worker/gpu_model_runner.py

```diff
@@ -320,13 +319,12 @@ def _prepare_inputs(self, scheduler_output: "SchedulerOutput"):
         # request in the batch. While we should not sample any token from this
         # partial request, we do so for simplicity. We will ignore the sampled
         # token from the partial request.
-        # TODO: Support prompt logprobs.
-        logits_indices = query_start_loc[1:] - 1
-        return input_ids, attn_metadata, logits_indices
+        return (input_ids, attn_metadata)
 
     def _prepare_sampling(
```

Comment:
Can we just execute the original `prepare_inputs` first and pass the `attn_metadata.query_start_loc` to `prepare_sampling`?

Reply:
Agreed; I made this change.